source/secure_tunneling.c

```diff
@@ -675,54 +1795,121 @@ struct aws_secure_tunnel *aws_secure_tunnel_new(const struct aws_secure_tunnel_o
 }
 
 struct aws_secure_tunnel *aws_secure_tunnel_acquire(struct aws_secure_tunnel *secure_tunnel) {
-    aws_ref_count_acquire(&secure_tunnel->ref_count);
+    if (secure_tunnel != NULL) {
+        aws_ref_count_acquire(&secure_tunnel->ref_count);
+    }
     return secure_tunnel;
 }
 
-void aws_secure_tunnel_release(struct aws_secure_tunnel *secure_tunnel) {
-    if (secure_tunnel == NULL) {
-        return;
+struct aws_secure_tunnel *aws_secure_tunnel_release(struct aws_secure_tunnel *secure_tunnel) {
+    if (secure_tunnel != NULL) {
+        aws_ref_count_release(&secure_tunnel->ref_count);
     }
-    aws_ref_count_release(&secure_tunnel->ref_count);
+
+    return NULL;
 }
 
-static void s_secure_tunnel_destroy(void *user_data) {
-    struct aws_secure_tunnel *secure_tunnel = user_data;
+int aws_secure_tunnel_start(struct aws_secure_tunnel *secure_tunnel) {
+    return s_aws_secure_tunnel_change_desired_state(secure_tunnel, AWS_STS_CONNECTED);
+}
 
-    aws_secure_tunneling_on_termination_complete_fn *on_termination_complete = NULL;
-    void *termination_complete_user_data = NULL;
-    if (secure_tunnel->options != NULL) {
-        on_termination_complete = secure_tunnel->options->on_termination_complete;
-        termination_complete_user_data = secure_tunnel->options->user_data;
+int aws_secure_tunnel_stop(struct aws_secure_tunnel *secure_tunnel) {
```

Comment:
Minor: Add `AWS_PRECONDITION(secure_tunnel != NULL);`?

Reply:
There are fatal asserts as soon as the state change is initiated that should cover this scenario. I think adding it here might be a needless double dip on that check.